### DISCUSSION TITLE
Where on first/latest of 1:n relation

### DISCUSSION THREAD

**Discussion by riccardolardi** [prisma/prisma#24890] [Q&A]
### Question

I can't seem to find a solution to this problem: I would like to do a findMany query and get only those entries whose **first (latest) relation of many (1:n) meets a condition**, eg. a field equals to some value. I'm using latest Prisma with Postgres. Is this not possible? Should I do a rawQuery? Thanks

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by riccardolardi** [comment]
Bump... pseudo code could look like this:

```ts
{
  where: {
    myRelation: {
      first: {
        myField: {
          equals: "some value"
        }
    }
  }
}
```

**Comment by ludralph** [comment]
Hi @riccardolardi 👋 

To help you better with your question, could you please provide your Prisma schema file or at least a few relevant models with a concrete example on what they are trying to achieve? It would be helpful to see how your relations are set up and understand the specific condition you want to filter by.

**Reply by riccardolardi** [reply]
@ludralph thanks for getting back.

I basically just want to filter depending on the contents of the "first" entry of `myRelation`:

```js
{
  // ...
  where: {
    myRelation: {
      take: 1, // this is not possible for relations in where clause it seems
      myField: {
        equals: "some value"
      }
    }
  }
}
```

**Reply by riccardolardi** [reply]
Hi @ludralph does this make sense? I'm still not sure whether this is a limitation in Prisma. I need to filter by a field on the latest (one single) relation entry.

**Reply by ludralph** [reply]
Hi,

I understand you're looking for a way to filter records based on the first (or latest) related record in a one-to-many relationship. Prisma doesn't offer a first or latest filter option within relation queries. Using a raw SQL query might be the most straightforward solution for your use-case.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
